google-cloud-datastore/src/main/java/com/google/cloud/datastore/testing/LocalDatastoreHelper.java

```diff
@@ -113,11 +115,95 @@ private LocalDatastoreHelper(double consistency, int port) {
     emulatorRunners = ImmutableList.of(gcloudRunner, downloadRunner);
   }
 
+  /** A builder for {@code LocalDatastoreHelper} objects. */
+  public static class Builder {
+    private List<EmulatorRunner> emulatorRunners;
+    private double consistency;
+    private int port;
+    private Path dataDir;
+    private boolean storeOnDisk = Boolean.TRUE.booleanValue();
+
+    private Builder() {}
+
+    public Builder setConsistency(double consistency) {
+      this.consistency = consistency;
+      return this;
+    }
+
+    public Builder setPort(int port) {
+      this.port = port;
+      return this;
+    }
+
+    public Builder setDataDir(Path dataDir) {
+      this.dataDir = dataDir;
+      return this;
+    }
+
+    public Builder setStoreOnDisk(boolean storeOnDisk) {
+      this.storeOnDisk = storeOnDisk;
+      return this;
+    }
+
+    /** Creates a {@code LocalDatastoreHelper} object. */
+    public LocalDatastoreHelper build() {
+      return new LocalDatastoreHelper(this);
+    }
+  }
+
+  LocalDatastoreHelper(Builder builder) {
+    super(
+        "datastore",
+        builder.port > 0 ? builder.port : BaseEmulatorHelper.findAvailablePort(DEFAULT_PORT),
+        PROJECT_ID_PREFIX + UUID.randomUUID().toString());
+    this.consistency = builder.consistency > 0 ? builder.consistency : DEFAULT_CONSISTENCY;
+    this.port = builder.port;
```

Comment:
just wondering: we pass the builder.port to the super and store the port in this class. Is it normal?

Reply:
thanks for catching this here, we need to call `getPort()` method of super.